src/components/block/block.ts

```diff
@@ -0,0 +1,320 @@
+		this._element = this._createDocumentElement(tagName);
+		this._addAttributes();
+	}
+
+	_addAttributes(): void {
+		// customize your this._element here
+	}
+
+	init(): void {
+		this._createResources();
+		this._render();
+		this.eventBus.emit(Block.EVENTS.FLOW_CDM);
+	}
+
+	_componentDidMount(): void {
+		this.componentDidMount();
+	}
+	
+	componentDidMount(): void {
+		// add did mount logic of your component here
+	}
+
+	dispatchComponentDidMount(): void {
+		this.componentDidMount();
+	}
+
+	_componentDidUpdate(oldProps: TProps, newProps: TProps): void {
+		const isUpdated = this.componentDidUpdate(oldProps, newProps);
+		if (isUpdated) {
+			this.eventBus.emit(Block.EVENTS.FLOW_RENDER);
+		}
+	}
+
+	componentDidUpdate(oldProps: TProps, newProps: TProps): boolean {
+		const oldKeys = Object.keys(oldProps);
+		const newKeys = Object.keys(newProps);
+		if(oldKeys.length !== newKeys.length) {
+			return true;
+		}
+		return oldKeys.reduce((acc, key) => {
+			return acc || oldProps[key] !== newProps[key];
+		}, false);
+	}
+
+	getProps = (): TProps => {
+		return this.props;
+	}
+
+	setProps = (nextProps: TProps): void => {
+		if (!nextProps) {
+			return;
+		}
+		const oldProps = {
+			...this.props,
+		}
+		Object.assign(this.props, nextProps);
+		this.eventBus.emit(Block.EVENTS.FLOW_CDU, oldProps, this.props);
+	};
+
+	get id(): string {
+		return this._id;
+	}
+
+	get element(): HTMLElement {
+		return this._element as HTMLElement; //safe assert since constructor filled _element property
+	}
+
+	get eventBus(): EventBus {
+		return this._eventBus;
+	}
+
+	_render(): void {
+		const block = this.render();
+		this.element.innerHTML = '';
+		this.element.appendChild(block);
+		this._addListners();
+	}
+
+	// add your render logic here
+	render(): DocumentFragment {
+		const renderProps = { ...this.props }
+
+		this.children.forEach((item: TChild): void => {
+			renderProps[item.key] = `<div data-stub-id="${item.value.id}"></div>`
+		});
+
+		this._removeListners();
+
+		const fragment = this._createDocumentElement<HTMLTemplateElement>('template');
+		fragment.innerHTML = this._templateRender(renderProps);
+
+		this.children.forEach((item: TChild): void => {
+			const stub = fragment.content.querySelector(`[data-stub-id="${item.value.id}"]`);
+			if (!stub) {
+				console.error(`Can't find stub with id ${item.value.id}`);
+				return;
+			}
+			// item.value.setProps(this.childrenProps[item.key]);
+			stub.replaceWith(item.value.getContent());
+		});
+
+		return fragment.content;
+	}
+
+	getContent(): HTMLElement {
+		return this.element;
+	}
+
+	_makePropsProxy(props: Record<string, string | number>): Record<string, string | number> {
+		const eventBus = this.eventBus;
+		this.proxyData = new Proxy(props, {
+			set(target: TProps, prop: keyof TProps, value: string): boolean {
+				const oldProps = {...target};
+				target[prop] = value;
+				eventBus.emit(Block.EVENTS.FLOW_CDU, oldProps, target);
```

Comment:
Можно использовать стрелочные функции, тогда не придется прибегать к замыканию
```suggestion
		this.proxyData = new Proxy(props, {
			set: (target: TProps, prop: keyof TProps, value: string): boolean => {
				const oldProps = {...target};
				target[prop] = value;
				this.eventBus.emit(Block.EVENTS.FLOW_CDU, oldProps, target);
```